src/adapters/monitors/cloudflare.rs

```diff
@@ -25,29 +27,40 @@ pub struct CloudFlareMonitor {
     last_query_time: DateTime<Utc>,
 }
 
-impl CloudFlareMonitor {
+impl CloudflareMonitor {
     pub fn new(client: Client) -> Self {
         Self {
             client,
             control_version_id: None,
             canary_version_id: None,
             start_time: Utc::now(),
-            last_query_time: Utc::now() - Duration::minutes(5),
+            // Start the first query 5 mins early to get some extra baseline data
+            // 5 + 3 = 8 to account for Cloudflare's ~2 mins metrics delay
+            last_query_time: Utc::now() - Duration::minutes(8),
         }
     }
 }
 
 #[async_trait]
-impl Monitor for CloudFlareMonitor {
+impl Monitor for CloudflareMonitor {
     type Item = CategoricalObservation<5, ResponseStatusCode>;
 
+    fn get_config(&self) -> MonitorConfig {
+        MonitorConfig::CloudflareWorkersObservability {
+            account_id: self.client.account_id().clone(),
+            worker_name: self.client.worker_name().clone(),
+        }
+    }
+
     async fn query(&mut self) -> Result<Vec<Self::Item>> {
         info!("Querying Cloudflare for new metrics.");
         // This function queries the metrics that we care most about (2xx, 4xx, and 5xx errors),
         // compiles them into a list, then generates the correct number of
         // CategoricalObservations for each response code
         let utc_now = Utc::now();
-        let end_query_time: DateTime<Utc> = Utc::now();
+        // Cloudflare observability metrics take ~2 mins (according to the dashboard) to become available,
+        // so we actually need to start our query a few minutes before the current time to ensure we get all the data.
+        let end_query_time: DateTime<Utc> = utc_now - Duration::minutes(3);
```

Comment:
This comments are really great! Normally I'd also push these values into constants to avoid the "magic numbers" anti-pattern but the comments are very clear and well written, so I'd say it's not needed for now.